### ISSUE TITLE
Loss get NAN

### ISSUE THREAD

**Issue by kwangsooshin** [imatge-upc/skiprnn-2017-telecombcn#2]
Hi. I read your paper very impressively. I want to use your code. However, I have a problem using your code. Below is my code.

```python
class SkipGruModel(models.BaseModel):

    def create_model(self, model_input, vocab_size, num_frames, is_training=True, **unused_params):

        stacked_GRU = MultiSkipGRUCell([gru_size] * number_of_layers)

        loss = 0.0
        with tf.variable_scope("RNN"):
            outputs, state = tf.nn.dynamic_rnn(stacked_GRU, model_input,
                                               sequence_length=num_frames,
                                               dtype=tf.float32)

        aggregated_model = getattr(video_level_models,
                                   FLAGS.video_level_classifier_model)

        return aggregated_model().create_model(
            model_input=state[-1].h,
            vocab_size=vocab_size,
            is_training=is_training,
            **unused_params)
```
The data comes in a number of 1024 one-dimensional vectors (not fixed). I want to get the probability of each label. Therefore, I used the cross entropy loss function. 

However, NAN is output as the value of Loss at first step.

What should i do?

**Comment by victorcampos7** [comment]
It's hard to tell from this code snippet only, but it looks like you are not using the output of SkipGRU at all. You should be returning and computing your loss based on `outputs`.

You also need to split the actual outputs and the binary update gate in SkipGRU, as in the code sample in the README:

```python
# Dynamic RNN unfolding
rnn_outputs, rnn_states = tf.nn.dynamic_rnn(cell, x, dtype=tf.float32, initial_state=initial_state)

# Split the output into the actual RNN output and the state update gate
rnn_outputs, updated_states = rnn_outputs.h, rnn_outputs.state_gate
```

**Comment by kwangsooshin** [comment]
Hmm... Below is LSTM code which same dataset.

```python
class LstmModel(models.BaseModel):

  def create_model(self, model_input, vocab_size, num_frames, **unused_params):
    """Creates a model which uses a stack of LSTMs to represent the video.

    Args:
      model_input: A 'batch_size' x 'max_frames' x 'num_features' matrix of
                   input features.
      vocab_size: The number of classes in the dataset.
      num_frames: A vector of length 'batch' which indicates the number of
           frames for each video (before padding).

    Returns:
      A dictionary with a tensor containing the probability predictions of the
      model in the 'predictions' key. The dimensions of the tensor are
      'batch_size' x 'num_classes'.
    """
    lstm_size = FLAGS.lstm_cells
    number_of_layers = FLAGS.lstm_layers

    stacked_lstm = tf.contrib.rnn.MultiRNNCell(
            [
                tf.contrib.rnn.BasicLSTMCell(
                    lstm_size, forget_bias=1.0)
                for _ in range(number_of_layers)
                ])

    loss = 0.0

    outputs, state = tf.nn.dynamic_rnn(stacked_lstm, model_input,
                                       sequence_length=num_frames,
                                       dtype=tf.float32)

    aggregated_model = getattr(video_level_models,
                               FLAGS.video_level_classifier_model)

    return aggregated_model().create_model(
        model_input=state[-1].h,
        vocab_size=vocab_size,
        **unused_params)
```

I want to replace LSTM to SkipLSTM or SkipGRU. But It's not work just change them. How to adjust this code to put your SkipRNNs?

**Comment by victorcampos7** [comment]
Seeing this, the graph construction in your first comment looks fine. If you get NaNs, you should debug the different parts of the model to find where and why the error is happening.

**Comment by kwangsooshin** [comment]
Okay. Thanks. I think this was a problem with the tensor floor version. I read your paper very impressively and thank you for your comment.
